code/modules/mob/living/carbon/human/appearance.dm

```diff
@@ -12,8 +12,11 @@
 	if(!(new_species in all_species))
 		return
 
-	set_species(new_species)
-	var/datum/antagonist/antag = mind && player_is_antag(mind)
+	var/datum/antagonist/antag = get_antag_data(mind?.special_role)
+	if(!(new_species in antag.valid_species))
+		return
+	else
+		set_species(new_species)
```

Comment:
We're null checking mind prior, but if mind is null here, `antag` will also be null. We're now not checking for a null in `antag` where it's possible it will be. Also redundant else as prior branch returned
```suggestion
	var/datum/antagonist/antag = get_antag_data(mind?.special_role)
	if(!(new_species in antag?.valid_species))
		return
	set_species(new_species)
```